src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingGraveyard.java

```diff
@@ -1,539 +1,537 @@
+    private static final String GRAVEYARD = "graveyard";
+
+    /**
+     * Descriptive string of the profession.
+     */
+    private static final String UNDERTAKER = "undertaker";
+
+    /**
+     * The maximum building level of the hut.
+     */
+    private static final int MAX_BUILDING_LEVEL = 5;
+
+    /**
+     * NBTTag to store the grave.
+     */
+    private static final String TAG_GRAVES = "graves";
+
+    /**
+     * NBTTag to store grave data.
+     */
+    private static final String TAG_GRAVE_DATA = "gravedata";
+
+    /**
+     * NBTTag to store the visual grave positions.
+     */
+    private static final String TAG_VISUAL_GRAVES = "visualgraves";
+
+    /**
+     * NBTTag to store the visual grave positions blockpos.
+     */
+    private static final String TAG_VISUAL_GRAVES_BLOCKPOS = "visualgravesblockpos";
+
+    /**
+     * NBTTag to store the visual grave facing.
+     */
+    private static final String TAG_VISUAL_GRAVES_FACING = "visualgravesfacing";
+
+    /**
+     * NBTTag to store the grave BlockPos.
+     */
+    private static final String TAG_GRAVES_BLOCKPOS = "gravesPos";
+
+    /**
+     * The last field tag.
+     */
+    private static final String TAG_CURRENT_GRAVE = "currentGRAVE";
+
+    /**
+     * The tag to store the list of resting citizen in this graveyard
+     */
+    private static final String TAG_RIP_CITIZEN_LIST = "ripCitizenList";
+
+    /**
+     * The list of resting citizen in this graveyard.
+     */
+    private final List<String> restingCitizen = new ArrayList<>();
+
+    /**
+     * The grave the undertaker is currently collecting.
+     */
+    @Nullable
+    private BlockPos currentGrave;
+
+    /**
+     * The data of the last grave dug by the undertaker.
+     */
+    @Nullable
+    private GraveData lastGraveData;
+
+    /**
+     * Whether we did init tags
+     */
+    private boolean initTags = false;
+
+    /**
+     * Grave positions
+     */
+    private List<Tuple<BlockPos, Direction>> visualGravePositions = new ArrayList<>();
+
+    /**
+     * The graveyard random.
+     */
+    private Random random = new Random();
+
+    /**
+     * Public constructor which instantiates the building.
+     *
+     * @param c the colony the building is in.
+     * @param l the position it has been placed (it's id).
+     */
+    public BuildingGraveyard(final IColony c, final BlockPos l)
+    {
+        super(c, l);
+        keepX.put(itemStack -> ItemStackUtils.hasToolLevel(itemStack, ToolType.SHOVEL, TOOL_LEVEL_WOOD_OR_GOLD, getMaxToolLevel()), new Tuple<>(1, true));
+    }
+
+    /**
+     * Clear the current grave the undertaker is currently working on.
+     */
+    public void ClearCurrentGrave()
+    {
+        this.currentGrave = null;
+    }
+
+    /**
+     * Retrieves a random grave to work on for the undertaker.
+     *
+     * @return a field to work on.
+     */
+    @Nullable
+    public BlockPos getGraveToWorkOn()
+    {
+        if(currentGrave != null)
+        {
```

Comment:
Only this part is new, seems the file was missing some formatting. I've added that it checks for loading here, and releases the lock on the grave when its not loaded